src/screens/ContactScreen/index.js

```diff
@@ -404,16 +467,28 @@ const ContactScreen = ({navigation}) => {
         />
       )}
 
-      {isSearchMode && (
-        <SearchRecyclerView
-          text={debouncedText}
-          followed={followed}
-          selectedUsers={selectedUsers}
-          usernames={usernames}
-          setLoading={setIsLoadingSearchResult}
-          onHandleSelected={(value) => handleSelected(value)}
+      {isRecyclerViewShownSearch && usersSearch.length > 0 && !isLoadingSearchResult && (
+        <RecyclerListView
+          style={styles.recyclerview}
+          layoutProvider={layoutProviderSearch}
+          dataProvider={dataProviderSearch}
+          extendedState={{
+            followed
+          }}
+          rowRenderer={rowRendererSearch}
+          scrollViewProps={{
+            refreshControl: (
+              <RefreshControl
+                tintColor={COLORS.white}
+                refreshing={refreshing}
+                onRefresh={onRefresh}
+              />
+            )
+          }}
         />
       )}
+
+      {isLoadingSearchResult && <LoadingWithoutModal />}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The RecyclerListView for displaying search results is introduced. It uses the new state variables and functions related to search. The implementation looks correct. However, as mentioned earlier, consider encapsulating the search-related states and functions within a custom hook or context to improve modularity and maintainability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->